src/Visitor/Php/Symfony/Constraint.php

```
@@ -0,0 +1,173 @@
+        'NotCompromisedPassword',
+        'NotEqualTo',
+        'NotIdenticalTo',
+        'NotNull',
+        'NumberConstraintTrait',
+        'Optional',
+        'Positive',
+        'PositiveOrZero',
+        'Range',
+        'Regex',
+        'Required',
+        'Time',
+        'Timezone',
+        'Traverse',
+        'Type',
+        'Unique',
+        'Url',
+        'Uuid',
+        'Valid',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function beforeTraverse(array $nodes): ?Node
+    {
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function enterNode(Node $node): ?Node
+    {
+        if (!$node instanceof Node\Expr\New_) {
+            return null;
+        }
+
+        $className = $node->class;
+        if (!$className instanceof Node\Name) {
+            return null;
+        }
+
+        $parts = $className->parts;
+        $isConstraintClass = false;
+        foreach ($parts as $part) {
+            if (\in_array($part, self::CONSTRAINT_CLASS_NAMES)) {
+                $isConstraintClass = true;
+            }
+        }
+
+        if (!$isConstraintClass) {
+            return null;
+        }
+
+        $args = $node->args;
+        if (0 === \count($args)) {
+            return null;
+        }
+
+        $arg = $args[0];
+        if (!$arg instanceof Node\Arg) {
+            return null;
+        }
+
+        $options = $arg->value;
+        if (!$options instanceof Node\Expr\Array_) {
+            return null;
+        }
+
+        $message = null;
+        $messageNode = null;
+
+        foreach ($options->items as $item) {
+            if (!$item->key instanceof Node\Scalar\String_) {
+                continue;
+            }
+
+            if ('message' !== $item->key->value) {
+                continue;
+            }
+
+            if (!$item->value instanceof Node\Scalar\String_) {
+                $this->addError($item, 'Constraint message is not a scalar string');
```

Do you think we need to add an error about this? Could you explain why it's better than just silently ignore non-string values?

---

I don't know, I like to get errors when translating so that I know that the extractor failed on one part but if you want I can just silently ignore it!